schema/deploy/shipping/views.sql

```diff
@@ -740,24 +754,316 @@ comment on view shipping.return_results_v2 is
+                             'pregnant_yesno',
+                             'income',
+                             'housing_type',
+                             'house_members',
+                             'doctor_3e8fae',
+                             'hospital_where',
+                             'hospital_ed',
+                             'hospital_arrive',
+                             'hospital_leave',
+                             'smoke_9a005',
+                             'chronic_illness',
+                             'overall_risk_health',
+                             'overall_risk_setting',
+                             'longterm_type')
+          group by encounter_id, "linkId"
+        ),
+
+        scan_study_arm as (
+          select encounter_id,
+                 string_response[1] as scan_study_arm
+            from questionnaire_responses
+          where "linkId" = 'redcap_event_name'
+        ),
+
+        vaccine as (
+          select encounter_id,
+                 boolean_response as vaccine,
+                 date_response[1] as vaccine_date
+            from questionnaire_responses
+          where "linkId" = 'vaccine'
+        ),
+
+        race as (
+          select encounter_id,
+                 case
+                    when array_length(code_response, 1) is null then string_response
+                    else code_response
+                 end as race
+            from questionnaire_responses
+          where "linkId" = 'race'
+        ),
+
+        insurance as (
+          select encounter_id,
+                 string_response as insurance
+            from questionnaire_responses
+          where "linkId" = 'insurance'
+        ),
+
+        ethnicity as (
+          select encounter_id,
+                 boolean_response as hispanic_or_latino
+            from questionnaire_responses
+          where "linkId" = 'ethnicity'
+        ),
+
+        travel_countries as (
+          select encounter_id,
+                 bool_or(boolean_response) as travel_countries
+            from questionnaire_responses
+          where "linkId" in ('travel_countries','travel_countries_phs')
+          group by encounter_id
+        ),
+
+        travel_states as (
+          select encounter_id,
+                 bool_or(boolean_response) as travel_states
+            from questionnaire_responses
+          where "linkId" in ('travel_states', 'travel_countries_phs')
+          group by encounter_id
+        ),
+
+        pregnant as (
+          select encounter_id,
+                 boolean_response as pregnant
+            from questionnaire_responses
+          where "linkId" = 'pregnant_yesno'
+        ),
+
+        income as (
+          select encounter_id,
+                 string_response as income
+            from questionnaire_responses
+          where "linkId" = 'income'
+        ),
+
+        housing_type as (
+          select encounter_id,
+                 string_response as housing_type
```

Comment:
This is a single-choice field, so I think would safe to grab just the first element and make analysis easier.

Some other string fields are the same way, e.g. house_members and clinical_care. I didn't check others, but would be good to. Some of this is clear from looking at existing data (e.g.  house_members), others have to look at REDCap field type.

Think this is worth the extra work to check?

Reply:
Yup, will update!